whylog/config/rule.py

```diff
@@ -32,6 +33,29 @@ def get_causes_parsers(self):
     def get_effect_name(self):
         return self._effect.name
 
+    def _gather_causes_frequency_information(self):
+        causes_with_frequency_info = [(self._causes[0].name, 0)]
+        for cause in self._causes:
```

Comment:
Did you assume that self._causes is ordered by cause.name? Now is unsorted. You can get sorted values, but you can't change self._causes, because you destroy clues_groups mapping in rule's constraints. If you decide to sorted replace tuples for two elements lists. Its avoid creating many objects. In my opinion the best solution is usage of ```python from collections import Counter ```